Allow arguments to be quickly extended with a hotkey

JadenTepper opened the pull request:
Hey! I judged at the Lexington tournament for LD a few weeks ago and tested two new features that I find pretty important to me. Here is one of the two.

## Motivation
Showing that an argument is extended (carried through to the next speech) can be difficult, especially when it goes unresponded.

## Fix
I made `control/cmd + e` extend the hovered argument to the next speech. There is also a button top right that can be used to activate the same feature. 

Note: I also added an option that makes tab return to the parent node on overrun because I prefer this behavior. It is default off, and likely change anything when off. 

##
Implementation Details
1. I made a new flag for `Box` nodes indicating if it is an extension. If it is an extension, it can't be edited and displays a right arrow.
2. Logic was added to ensure responses are not added on top of the extension node.
3. Logic was created to not allow extensions of extensions or double extensions.
4. The extension button in the top right can be disabled (it is currently not grouped with any of the other disable options).
5. I tested the feature at the tournament, and it worked well for me, but I may be missing edge cases.

## Needed Changes
1. I didn't know where you got your button symbols, so I temporarily used the `eye` symbol.
2. I used a right arrow character for the extension node for simplicity, but the symbol for an extension can be changed.

Let me know what you think, and if anything should be changed.

Ashwagandhae commented:
Hi! Thank you so much for this PR, I just tried it out and it's very cool! I've been wanting to add something like this for a while and your implementation is very impressive. Using a shortcut to extend an argument feels very natural and fits well with the rest of Flower. 

Some bugs: 
- Currently, you have to undo twice to undo one extension. Instead of calling `addNewBox` twice when creating an extension cell, create a new function in `nodeDecorateAction.ts` that puts two `add` actions into an ActionBundle, to create proper behaviour when undoing. 

Some changes I would make:
- I think that the extension cell should keep the color of column instead of getting the opposite color:
   - It doesn't mess up the new cell button in the next column
   - It keeps it clear whether you're currently on an aff column or neg column—this could get confusing if there are a lot of extensions
- Extension cells should probably not be boldable/strikethroughable
- Although keeping the textarea there to keep focus state correct is good, it probably shouldn't be used to display icons. Instead, I think conditionally overlaying an `Icon` element and keeping text empty would be a cleaner solution. 

Some questions: 
- Why is adding arguments above extensions not allowed, but adding arguments below is?
- I agree with you that there should be icons for both the button symbols and the extension node (probably the same icon). What would be a good icon? A normal `→` arrow? An implies arrow to reduce confusion `⇒`? A double arrow `⇇`? Let me know what you think, I can add the icon to the icon set (they're made by me in figma). 

Thank you again for all the work you put into this PR! Let me know which of these changes you'd like to tackle—I’m happy to make any of them if needed!

JadenTepper commented:
Thanks for the feedback! I can handle all the implementation changes except for drawing the icon.

I originally disallowed placing responses above extensions because it caused the extension to become misaligned with its parent argument, which I felt looked confusing. When responses are placed above, the extension box no longer aligns with the original argument, and the arrow can appear to point from nothing. However, if you think it's more functional without this rule, I'm happy to adjust it.

Ashwagandhae commented:
I've added a `arrowRightThroughCircle` to represent extensions, and pushed it to your PR. You should be able to create an `<Icon name="arrowRightThroughCircle">`. I chose an arrow going through a circle to make it significantly different from the undo/redo arrows, and to have the circle represent the emptiness/zeroness of the opponent not responding to the argument. 

I see why you disallowed placing responses above, you can leave it as is!

JadenTepper commented:
I've incorporated your suggestions and tested the changes—everything seems to be working, but let me know if there's anything else that needs tweaking. I also fixed a bug where Excel exports would fail if a flow was left unnamed.

I also found another small bug when testing. On Mac Safari (at least on my machine), when a box is created it is 1px taller than it should be. It gets updated when there is an input event. However, extensions, which never get an input event, don't get fixed. I couldn't figure out the root cause though. Let me know if I should look more into this.

Ashwagandhae reviewed (commented):
Looks good!

Ashwagandhae reviewed (approved):
Looks good!

Ashwagandhae commented:
Thank you so much for all the work you put in this! I've decided to merge regardless of the Safari bug because it doesn't impact usability that much, but let me know if you find a fix for it.
